qiskit/optimization/converters/quadratic_program_converter.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An abstract class for optimization algorithms in Qiskit's optimization module."""
+
+from abc import ABC, abstractmethod
+
+from ..algorithms.optimization_algorithm import OptimizationResult
+from ..problems.quadratic_program import QuadraticProgram
+
+
+class QuadraticProgramConverter(ABC):
```

Comment:
As 'name' seems to be a common aspect would it make sense to have a constructor here that took and stored the name so this class would have the setter/getter for name too. I see a lot of setter/getter pairs for name hence wondering if more common function can go to this class.

Reply:
I think we need to discuss about `name` again. This seems to be an exception to the rule that all parameters should be moved to `__init__`. The given `name` is the name of the newly created `QuadraticProgram`. Thus, setting this as a parameter of the converter object doesn't seem to make sense (all problems would get the same name). It should be an optional argument to `convert` that is `None` by default, which means the new problem has the same name as the old problem. This also implies that we don't need setters/getters for `name`.

Reply:
Because algorithms can modify the name of `QuadraticProgram`, it might be good not to touch name in converters to be simple.

Reply:
@t-imamichi good for me.